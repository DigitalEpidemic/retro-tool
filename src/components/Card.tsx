import { useState, useRef } from "react";
import { ThumbsUp, ThumbsDown, Edit2, Trash2 } from "lucide-react";
// Import from boardService instead of cardService
import { updateCard, deleteCard, voteForCard } from "../services/boardService";
import { DraggableProvided } from "@hello-pangea/dnd"; // Import DraggableProvided
import { Card as CardType } from "../services/firebase"; // Import Card type

interface CardProps {
  provided: DraggableProvided; // Use DraggableProvided type
  card: CardType; // Use CardType
  isOwner: boolean;
}

export default function Card({ provided, card, isOwner }: CardProps) {
  const [isEditing, setIsEditing] = useState(false);
  const [content, setContent] = useState(card.content);
  const [cardHeight, setCardHeight] = useState(0);
  const cardRef = useRef<HTMLDivElement>(null);

  // Function to determine card color based on column or random assignment
  const getCardColor = () => {
    // This is a simple mapping - extend as needed based on your column IDs
    const columnColors: Record<string, string> = {
      // Mad column (first column) - mint green
      "column-1": "bg-green-100",
      // Sad column (second column) - light purple for some cards, mint green for others
      "column-2":
        card.id.charCodeAt(0) % 2 === 0 ? "bg-green-100" : "bg-purple-100",
      // Glad column (third column) - alternating colors
      "column-3":
        card.id.charCodeAt(0) % 2 === 0 ? "bg-green-100" : "bg-purple-100",
      // Default fallback for any other columns
      default:
        card.id.charCodeAt(0) % 2 === 0 ? "bg-green-100" : "bg-purple-100",
    };

    // Return the color for the column or default to the fallback if column not found
    return columnColors[card.columnId] || columnColors["default"];
  };

  const handleSave = () => {
    if (content.trim()) {
      updateCard(card.id, { content });
      setIsEditing(false);
    }
  };

  const handleEdit = () => {
    if (cardRef.current) {
      const height = cardRef.current.clientHeight;
      setCardHeight(height);
    }
    setIsEditing(true);
  };

  const handleDelete = () => {
    if (window.confirm("Are you sure you want to delete this card?")) {
      deleteCard(card.id);
    }
  };

  const handleVote = (voteType: "up" | "down") => {
    voteForCard(card.id, voteType).catch((error) => {
      console.error("Vote failed:", error);
    });
  };

  return (
    <div
      ref={(node) => {
        provided.innerRef(node);
        cardRef.current = node;
      }}
      {...provided.draggableProps}
      {...provided.dragHandleProps}
      className={`${getCardColor()} rounded shadow-sm border-none mb-3 group p-0 min-h-[100px] relative flex flex-col overflow-hidden`}
    >
      {isEditing ? (
        <div className="p-3" style={{ height: `${cardHeight}px` }}>
          <textarea
            value={content}
            onChange={(e) => setContent(e.target.value)}
            className="w-full h-[calc(100%-40px)] rounded border border-gray-300 focus:border-blue-500 focus:ring-1 focus:ring-blue-500 text-sm p-2 resize-none"
            autoFocus
          />
          <div className="flex justify-end space-x-2 mt-2">
            <button
              aria-label="Cancel"
              onClick={() => setIsEditing(false)}
              className="px-2.5 py-1 text-xs font-medium text-gray-700 bg-gray-100 rounded hover:bg-gray-200 focus:outline-none focus:ring-2 focus:ring-gray-300 cursor-pointer"
            >
              Cancel
            </button>
            <button
              aria-label="Save"
              onClick={handleSave}
              className="px-2.5 py-1 text-xs font-medium bg-blue-500 text-white rounded hover:bg-blue-600 focus:outline-none focus:ring-2 focus:ring-blue-500 cursor-pointer"
            >
              Save
            </button>
          </div>
        </div>
      ) : (
        <>
          {/* Position edit/delete buttons in top right corner */}
          <div className="absolute right-0 top-0 flex flex-col items-center py-2 px-0 min-w-[30px] z-10">
            {isOwner && (
              <>
                {/* Delete button at top */}
                <button
                  aria-label="Delete"
                  onClick={handleDelete}
                  className="p-1 rounded text-red-600 hover:text-red-700 hover:bg-red-100 transition-colors cursor-pointer"
                >
                  <Trash2 className="h-3 w-3" />
                </button>

                {/* Edit button below delete */}
                <button
                  aria-label="Edit"
                  onClick={handleEdit}
                  className="p-1 mt-2 rounded text-blue-600 hover:text-blue-700 hover:bg-blue-100 transition-colors cursor-pointer"
                >
                  <Edit2 className="h-3 w-3" />
                </button>
              </>
            )}
          </div>

          {/* Card content with padding to accommodate action buttons */}
          <div className="flex-grow py-3 mr-[30px] ml-[30px]">
            <p className="text-sm text-gray-900 whitespace-pre-wrap break-words leading-relaxed text-left">
              {card.content}
            </p>
          </div>

          {/* Footer with author name and voting */}
          <div className="px-3 pb-2 relative">
            <div className="flex items-center justify-between">
              {/* Voting component horizontal layout on left */}
              <div className="flex items-center">
                <button
                  onClick={() => handleVote("up")}
                  className="p-1 rounded flex items-center transition-colors cursor-pointer text-gray-500 hover:text-green-600 hover:bg-green-50"
                >
                  <ThumbsUp aria-label="Upvote" className="h-3 w-3" />
                </button>

                <span className="text-xs font-medium mx-1">{card.votes}</span>

                <button
                  onClick={() => handleVote("down")}
                  className="p-1 rounded flex items-center transition-colors cursor-pointer text-gray-500 hover:text-red-600 hover:bg-red-50"
                >
                  <ThumbsDown aria-label="Downvote" className="h-3 w-3" />
                </button>
              </div>

              {/* Author name on right */}
              <div className="text-right">
                <div className="text-xs text-gray-500 italic">
                  {card.authorName || "Wonderful Turtle"}
                </div>
              </div>
            </div>
          </div>
        </>
      )}
    </div>
  );
}
