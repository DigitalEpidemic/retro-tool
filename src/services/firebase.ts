import { initializeApp } from "firebase/app";
import { getAuth } from "firebase/auth";
import { getDatabase } from "firebase/database";
import { getFirestore, Timestamp } from "firebase/firestore";
import { ActionPoint } from "../components/ActionPointsPanel";

// Your web app's Firebase configuration
// In a real app, set this up as environment variables (.env file)
const firebaseConfig = {
  apiKey: import.meta.env.VITE_FIREBASE_API_KEY,
  authDomain: import.meta.env.VITE_FIREBASE_AUTH_DOMAIN,
  projectId: import.meta.env.VITE_FIREBASE_PROJECT_ID,
  storageBucket: import.meta.env.VITE_FIREBASE_STORAGE_BUCKET,
  messagingSenderId: import.meta.env.VITE_FIREBASE_MESSAGING_SENDER_ID,
  appId: import.meta.env.VITE_FIREBASE_APP_ID,
  databaseURL:
    import.meta.env.VITE_FIREBASE_DATABASE_URL ||
    `https://${
      import.meta.env.VITE_FIREBASE_PROJECT_ID
    }-default-rtdb.firebaseio.com`,
};

// Initialize Firebase
const app = initializeApp(firebaseConfig);
export const db = getFirestore(app);
export const auth = getAuth(app);
export const rtdb = getDatabase(app);

// Type for a board column
export interface Column {
  id: string;
  title: string;
  order: number;
  sortByVotes?: boolean;
}

// Type for a board
export interface Board {
  id: string;
  name: string;
  createdAt: Timestamp;
  isActive: boolean;
  columns: Record<string, Column>;
  facilitatorId?: string;
  timerIsRunning?: boolean;
  timerStartTime?: Timestamp;
  timerDurationSeconds?: number;
  timerPausedDurationSeconds?: number;
  timerOriginalDurationSeconds?: number;
  actionPoints?: ActionPoint[];
  showAddColumnPlaceholder?: boolean;
}

// Type for a card (note card)
export interface Card {
  id: string;
  boardId: string;
  columnId: string;
  content: string;
  authorId: string;
  authorName: string;
  createdAt: Timestamp;
  votes: number;
  position: number;
  actionable?: boolean;
  color?: string; // Tailwind class name (e.g., 'bg-red-200')
}

// Type for a user
export interface User {
  id: string;
  name: string;
  color: string; // Tailwind class name (e.g., 'bg-red-200')
  boardId: string;
  lastActive: Timestamp;
  isViewingPage?: boolean;
}

// Type for online user in realtime database
export interface OnlineUser {
  id: string;
  name: string;
  color: string; // Tailwind class name (e.g., 'bg-red-200')
  boardId: string;
  lastOnline: number;
}
