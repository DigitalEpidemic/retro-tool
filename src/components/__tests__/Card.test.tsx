import { DraggableProvided } from '@hello-pangea/dnd';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { Timestamp } from 'firebase/firestore'; // Import Timestamp
import { beforeEach, describe, expect, it, vi } from 'vitest';
import { Card as CardType } from '../../services/firebase';
import Card from '../Card';

// Mock the boardService functions
vi.mock('../../services/boardService', () => ({
  updateCard: vi.fn(),
  deleteCard: vi.fn(),
  voteForCard: vi.fn().mockResolvedValue(undefined), // Mock vote to resolve successfully
}));

// Mock icons to avoid SVG rendering issues in tests
vi.mock('lucide-react', () => ({
  Edit2: () => <div data-testid="edit-icon">Edit</div>,
  Trash2: () => <div data-testid="trash-icon">Delete</div>,
  Check: () => <div data-testid="check-icon">Save</div>,
  X: () => <div data-testid="x-icon">Cancel</div>,
  ThumbsUp: () => <div data-testid="thumbs-up-icon">Upvote</div>,
  ThumbsDown: () => <div data-testid="thumbs-down-icon">Downvote</div>,
}));

// Mock window.confirm
global.confirm = vi.fn(() => true); // Assume user confirms deletion

// Helper function to create mock DraggableProvided
const mockDraggableProvided = (): DraggableProvided => ({
  innerRef: vi.fn(),
  draggableProps: {
    // Add required properties based on TS errors
    'data-rfd-draggable-context-id': 'mock-context-1', // Using placeholder
    'data-rfd-draggable-id': 'mock-draggable-1', // Using placeholder
    style: {},
    onTransitionEnd: vi.fn(),
  },
  dragHandleProps: {
    // Add required properties based on TS errors
    'data-rfd-drag-handle-draggable-id': 'mock-draggable-1', // Using placeholder
    'data-rfd-drag-handle-context-id': 'mock-context-1', // Using placeholder
    role: 'button',
    tabIndex: 0,
    draggable: false,
    onDragStart: vi.fn(),
    'aria-describedby': 'id',
  },
});

describe('Card', () => {
  let mockCard: CardType;
  let provided: DraggableProvided;

  beforeEach(() => {
    // Reset mocks before each test
    vi.clearAllMocks();
    (global.confirm as ReturnType<typeof vi.fn>).mockClear().mockReturnValue(true);

    mockCard = {
      id: 'card-1',
      boardId: 'board-1', // Added
      content: 'Test Card Content',
      columnId: 'column-1', // Mad column (green)
      votes: 5,
      authorId: 'user-1',
      authorName: 'Test User',
      createdAt: Timestamp.now(), // Use Firestore Timestamp mock
      position: 1, // Added
    };
    provided = mockDraggableProvided();
  });

  // 1. Rendering Tests
  it('renders correctly with given props', () => {
    render(<Card provided={provided} card={mockCard} isOwner={true} />);
    expect(screen.getByText('Test Card Content')).toBeInTheDocument();
    expect(screen.getByText('5')).toBeInTheDocument(); // Votes
    expect(screen.getByText('Test User')).toBeInTheDocument(); // Author name
    expect(screen.getByRole('button', { name: 'Edit' })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Delete' })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Upvote' })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Downvote' })).toBeInTheDocument();
  });

  it('renders correctly when not the owner', () => {
    render(<Card provided={provided} card={mockCard} isOwner={false} />);
    expect(screen.getByText('Test Card Content')).toBeInTheDocument();
    expect(screen.queryByRole('button', { name: 'Edit' })).not.toBeInTheDocument();
    expect(screen.queryByRole('button', { name: 'Delete' })).not.toBeInTheDocument();
  });

  it('renders default author name if authorName is missing', () => {
    // Adjust mock to satisfy CardType while testing the component's fallback logic
    // Ideally, CardType would have authorName?: string
    const cardWithoutAuthorName = {
      ...mockCard,
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      authorName: undefined as any, // Cast to bypass TS error for this specific test case
    };
    render(
      <Card
        provided={provided}
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        card={cardWithoutAuthorName as CardType} // Cast needed because of the 'any' above
        isOwner={true}
      />
    );
    expect(screen.getByText('Wonderful Turtle')).toBeInTheDocument();
  });

  it('applies correct background color based on columnId (column-1)', () => {
    const { container } = render(<Card provided={provided} card={mockCard} isOwner={true} />); // Removed extra closing parenthesis here
    // Find the main div using the ref assignment logic (or a data-testid if added)
    // Here we check the first child div which receives the classes
    const cardElement = container.firstChild as HTMLElement;
    expect(cardElement).toHaveClass('bg-green-100');
  });

  it('applies correct background color based on columnId (column-2, even ID)', () => {
    const cardCol2Even = { ...mockCard, columnId: 'column-2', id: 'card-2' }; // Even ASCII for '2'
    const { container } = render(<Card provided={provided} card={cardCol2Even} isOwner={true} />);
    const cardElement = container.firstChild as HTMLElement;
    expect(cardElement).toHaveClass('bg-purple-100');
  });

  it('applies correct background color based on columnId (column-2, odd ID)', () => {
    const cardCol2Odd = { ...mockCard, columnId: 'column-2', id: 'card-1' }; // Odd ASCII for '1'
    const { container } = render(<Card provided={provided} card={cardCol2Odd} isOwner={true} />);
    const cardElement = container.firstChild as HTMLElement;
    expect(cardElement).toHaveClass('bg-purple-100');
  });

  it('applies correct background color based on columnId (column-3, even ID)', () => {
    const cardCol3Even = { ...mockCard, columnId: 'column-3', id: 'card-2' }; // Even ASCII for '2'
    const { container } = render(<Card provided={provided} card={cardCol3Even} isOwner={true} />);
    const cardElement = container.firstChild as HTMLElement;
    expect(cardElement).toHaveClass('bg-purple-100');
  });

  it('applies correct background color based on columnId (column-3, odd ID)', () => {
    const cardCol3Odd = { ...mockCard, columnId: 'column-3', id: 'card-1' }; // Odd ASCII for '1'
    const { container } = render(<Card provided={provided} card={cardCol3Odd} isOwner={true} />);
    const cardElement = container.firstChild as HTMLElement;
    expect(cardElement).toHaveClass('bg-purple-100');
  });

  it('applies default background color for unknown columnId (odd ID)', () => {
    const cardUnknownCol = {
      ...mockCard,
      columnId: 'unknown-column',
      id: 'card-1',
    }; // Odd ASCII for '1'
    const { container } = render(<Card provided={provided} card={cardUnknownCol} isOwner={true} />);
    const cardElement = container.firstChild as HTMLElement;
    expect(cardElement).toHaveClass('bg-purple-100'); // Default fallback for odd
  });

  it("uses card's color property when available", () => {
    const cardWithColor = {
      ...mockCard,
      color: 'bg-blue-300', // Dark blue Tailwind class
    };
    const { container } = render(<Card provided={provided} card={cardWithColor} isOwner={true} />);
    const cardElement = container.firstChild as HTMLElement;
    expect(cardElement).toHaveClass('bg-blue-300'); // Should directly use the Tailwind class
  });

  // 2. User Interaction Tests
  it('enters edit mode when edit button is clicked', () => {
    render(<Card provided={provided} card={mockCard} isOwner={true} />);
    // Find button via the icon's test id
    const editButton = screen.getByRole('button', { name: 'Edit' });
    expect(editButton).toBeInTheDocument(); // Ensure button is found
    fireEvent.click(editButton!); // Use non-null assertion

    expect(screen.getByRole('textbox')).toBeInTheDocument();
    expect(screen.getByDisplayValue('Test Card Content')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Save' })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
    expect(screen.queryByText(mockCard.authorName)).not.toBeInTheDocument();
  });

  it('calls updateCard and exits edit mode when save button is clicked', async () => {
    const { updateCard } = await import('../../services/boardService');
    render(<Card provided={provided} card={mockCard} isOwner={true} />);

    // Enter edit mode
    const editButton = screen.getByRole('button', { name: 'Edit' });
    expect(editButton).toBeInTheDocument();
    fireEvent.click(editButton!);

    // Change content
    const textarea = screen.getByRole('textbox');
    fireEvent.change(textarea, { target: { value: 'Updated Content' } });

    // Click save
    fireEvent.click(screen.getByRole('button', { name: 'Save' }));

    // Assertions
    expect(updateCard).toHaveBeenCalledTimes(1);
    expect(updateCard).toHaveBeenCalledWith('card-1', {
      content: 'Updated Content',
    });

    // Wait for the state update to hide the textarea
    await waitFor(() => {
      expect(screen.queryByRole('textbox')).not.toBeInTheDocument();
    });
    // Check if the original content display is back (though it will show old content until prop updates)
    expect(screen.getByText('Test Card Content')).toBeInTheDocument();
  });

  it('does not call updateCard if content is empty when saving', async () => {
    const { updateCard } = await import('../../services/boardService');
    render(<Card provided={provided} card={mockCard} isOwner={true} />);

    const editButton = screen.getByRole('button', { name: 'Edit' });
    expect(editButton).toBeInTheDocument();
    fireEvent.click(editButton!);
    const textarea = screen.getByRole('textbox');
    fireEvent.change(textarea, { target: { value: '   ' } }); // Empty content
    fireEvent.click(screen.getByRole('button', { name: 'Save' }));

    expect(updateCard).not.toHaveBeenCalled();
    // Should remain in edit mode
    expect(screen.getByRole('textbox')).toBeInTheDocument();
  });

  it('exits edit mode without saving when cancel button is clicked', async () => {
    const { updateCard } = await import('../../services/boardService');
    render(<Card provided={provided} card={mockCard} isOwner={true} />);

    // Enter edit mode
    const editButton = screen.getByRole('button', { name: 'Edit' });
    expect(editButton).toBeInTheDocument();
    fireEvent.click(editButton!);

    // Change content (optional, just to ensure it doesn't save)
    fireEvent.change(screen.getByRole('textbox'), {
      target: { value: 'Temporary Edit' },
    });

    // Click cancel
    fireEvent.click(screen.getByRole('button', { name: 'Cancel' }));

    // Assertions
    expect(updateCard).not.toHaveBeenCalled();
    await waitFor(() => {
      expect(screen.queryByRole('textbox')).not.toBeInTheDocument();
    });
    expect(screen.getByText('Test Card Content')).toBeInTheDocument(); // Original content shown
  });

  it('calls deleteCard when delete button is clicked and confirmed', async () => {
    const { deleteCard } = await import('../../services/boardService');
    render(<Card provided={provided} card={mockCard} isOwner={true} />);

    const deleteButton = screen.getByRole('button', { name: 'Delete' });
    expect(deleteButton).toBeInTheDocument();
    fireEvent.click(deleteButton!);

    expect(global.confirm).toHaveBeenCalledTimes(1);
    expect(deleteCard).toHaveBeenCalledTimes(1);
    expect(deleteCard).toHaveBeenCalledWith('card-1');
  });

  it('does not call deleteCard when delete button is clicked and cancelled', async () => {
    const { deleteCard } = await import('../../services/boardService');
    (global.confirm as ReturnType<typeof vi.fn>).mockReturnValue(false); // Simulate user cancelling
    render(<Card provided={provided} card={mockCard} isOwner={true} />);

    const deleteButton = screen.getByRole('button', { name: 'Delete' });
    expect(deleteButton).toBeInTheDocument();
    fireEvent.click(deleteButton!);

    expect(global.confirm).toHaveBeenCalledTimes(1);
    expect(deleteCard).not.toHaveBeenCalled();
  });

  it('calls voteForCard with "up" when thumbs up is clicked', async () => {
    const { voteForCard } = await import('../../services/boardService');
    render(<Card provided={provided} card={mockCard} isOwner={true} />);

    const thumbsUpButton = screen.getByRole('button', { name: 'Upvote' });
    expect(thumbsUpButton).toBeInTheDocument();
    fireEvent.click(thumbsUpButton!);

    expect(voteForCard).toHaveBeenCalledTimes(1);
    expect(voteForCard).toHaveBeenCalledWith('card-1', 'up');
  });

  it('calls voteForCard with "down" when thumbs down is clicked', async () => {
    const { voteForCard } = await import('../../services/boardService');
    render(<Card provided={provided} card={mockCard} isOwner={true} />);

    const thumbsDownButton = screen.getByRole('button', { name: 'Downvote' });
    expect(thumbsDownButton).toBeInTheDocument();
    fireEvent.click(thumbsDownButton!);

    expect(voteForCard).toHaveBeenCalledTimes(1);
    expect(voteForCard).toHaveBeenCalledWith('card-1', 'down');
  });

  // 3. State/Prop Change Tests (implicitly covered by interaction tests, but can add specific ones)
  it('updates displayed content when card prop changes', () => {
    const { rerender } = render(<Card provided={provided} card={mockCard} isOwner={true} />);
    expect(screen.getByText('Test Card Content')).toBeInTheDocument();

    const updatedCard = { ...mockCard, content: 'New Content From Prop' };
    rerender(<Card provided={provided} card={updatedCard} isOwner={true} />);

    expect(screen.getByText('New Content From Prop')).toBeInTheDocument();
    expect(screen.queryByText('Test Card Content')).not.toBeInTheDocument();
  });

  it('updates displayed votes when card prop changes', () => {
    const { rerender } = render(<Card provided={provided} card={mockCard} isOwner={true} />);
    expect(screen.getByText('5')).toBeInTheDocument();

    const updatedCard = { ...mockCard, votes: 10 };
    rerender(<Card provided={provided} card={updatedCard} isOwner={true} />);

    expect(screen.getByText('10')).toBeInTheDocument();
    expect(screen.queryByText('5')).not.toBeInTheDocument();
  });

  // 4. Firestore Integration (Mocked) - Covered by interaction tests calling mocked service functions

  // 5. Edge Cases
  it('handles card with empty content correctly', () => {
    const emptyContentCard = { ...mockCard, content: '' };
    render(<Card provided={provided} card={emptyContentCard} isOwner={true} />);
    // Check that the container is there, but the specific content text isn't
    expect(screen.queryByText('Test Card Content')).not.toBeInTheDocument();
    // Check for author and votes to ensure the rest renders
    expect(screen.getByText('5')).toBeInTheDocument();
    expect(screen.getByText('Test User')).toBeInTheDocument();
  });

  // 6. Snapshot Testing
  it('matches snapshot when owner', () => {
    const { container } = render(<Card provided={provided} card={mockCard} isOwner={true} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot when not owner', () => {
    const { container } = render(<Card provided={provided} card={mockCard} isOwner={false} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot when editing', () => {
    const { container } = render(<Card provided={provided} card={mockCard} isOwner={true} />);
    const editButton = screen.getByRole('button', { name: 'Edit' });
    expect(editButton).toBeInTheDocument();
    fireEvent.click(editButton!);
    expect(container.firstChild).toMatchSnapshot();
  });
});
