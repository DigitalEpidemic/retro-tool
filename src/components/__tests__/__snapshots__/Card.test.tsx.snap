// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card Component > matches snapshot when editing 1`] = `
<div
  aria-describedby="id"
  class="bg-green-100 rounded shadow-sm border-none mb-3 group p-0 min-h-[100px] relative flex flex-col overflow-hidden"
  data-rfd-drag-handle-context-id="mock-context-1"
  data-rfd-drag-handle-draggable-id="mock-draggable-1"
  data-rfd-draggable-context-id="mock-context-1"
  data-rfd-draggable-id="mock-draggable-1"
  draggable="false"
  role="button"
  tabindex="0"
>
  <div
    class="p-3"
    style="height: 0px;"
  >
    <textarea
      class="w-full h-[calc(100%-40px)] rounded border border-gray-300 focus:border-blue-500 focus:ring-1 focus:ring-blue-500 text-sm p-2 resize-none"
    >
      Test Card Content
    </textarea>
    <div
      class="flex justify-end space-x-2 mt-2"
    >
      <button
        aria-label="Cancel"
        class="px-2.5 py-1 text-xs font-medium text-gray-700 bg-gray-100 rounded hover:bg-gray-200 focus:outline-none focus:ring-2 focus:ring-gray-300 cursor-pointer"
      >
        Cancel
      </button>
      <button
        aria-label="Save"
        class="px-2.5 py-1 text-xs font-medium bg-blue-500 text-white rounded hover:bg-blue-600 focus:outline-none focus:ring-2 focus:ring-blue-500 cursor-pointer"
      >
        Save
      </button>
    </div>
  </div>
</div>
`;

exports[`Card Component > matches snapshot when not owner 1`] = `
<div
  aria-describedby="id"
  class="bg-green-100 rounded shadow-sm border-none mb-3 group p-0 min-h-[100px] relative flex flex-col overflow-hidden"
  data-rfd-drag-handle-context-id="mock-context-1"
  data-rfd-drag-handle-draggable-id="mock-draggable-1"
  data-rfd-draggable-context-id="mock-context-1"
  data-rfd-draggable-id="mock-draggable-1"
  draggable="false"
  role="button"
  tabindex="0"
>
  <div
    class="absolute right-0 top-0 flex flex-col items-center py-2 px-0 min-w-[30px] z-10"
  />
  <div
    class="flex-grow py-3 mr-[30px] ml-[30px]"
  >
    <p
      class="text-sm text-gray-900 whitespace-pre-wrap break-words leading-relaxed text-left"
    >
      Test Card Content
    </p>
  </div>
  <div
    class="px-3 pb-2 relative"
  >
    <div
      class="flex items-center justify-between"
    >
      <div
        class="flex items-center"
      >
        <button
          class="p-1 rounded flex items-center transition-colors cursor-pointer text-gray-500 hover:text-green-600 hover:bg-green-50"
        >
          <svg
            aria-label="Upvote"
            class="lucide lucide-thumbs-up h-3 w-3"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 10v12"
            />
            <path
              d="M15 5.88 14 10h5.83a2 2 0 0 1 1.92 2.56l-2.33 8A2 2 0 0 1 17.5 22H4a2 2 0 0 1-2-2v-8a2 2 0 0 1 2-2h2.76a2 2 0 0 0 1.79-1.11L12 2a3.13 3.13 0 0 1 3 3.88Z"
            />
          </svg>
        </button>
        <span
          class="text-xs font-medium mx-1"
        >
          5
        </span>
        <button
          class="p-1 rounded flex items-center transition-colors cursor-pointer text-gray-500 hover:text-red-600 hover:bg-red-50"
        >
          <svg
            aria-label="Downvote"
            class="lucide lucide-thumbs-down h-3 w-3"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 14V2"
            />
            <path
              d="M9 18.12 10 14H4.17a2 2 0 0 1-1.92-2.56l2.33-8A2 2 0 0 1 6.5 2H20a2 2 0 0 1 2 2v8a2 2 0 0 1-2 2h-2.76a2 2 0 0 0-1.79 1.11L12 22a3.13 3.13 0 0 1-3-3.88Z"
            />
          </svg>
        </button>
      </div>
      <div
        class="text-right"
      >
        <div
          class="text-xs text-gray-500 italic"
        >
          Test User
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Card Component > matches snapshot when owner 1`] = `
<div
  aria-describedby="id"
  class="bg-green-100 rounded shadow-sm border-none mb-3 group p-0 min-h-[100px] relative flex flex-col overflow-hidden"
  data-rfd-drag-handle-context-id="mock-context-1"
  data-rfd-drag-handle-draggable-id="mock-draggable-1"
  data-rfd-draggable-context-id="mock-context-1"
  data-rfd-draggable-id="mock-draggable-1"
  draggable="false"
  role="button"
  tabindex="0"
>
  <div
    class="absolute right-0 top-0 flex flex-col items-center py-2 px-0 min-w-[30px] z-10"
  >
    <button
      aria-label="Delete"
      class="p-1 rounded text-red-600 hover:text-red-700 hover:bg-red-100 transition-colors cursor-pointer"
    >
      <svg
        class="lucide lucide-trash2 lucide-trash-2 h-3 w-3"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 6h18"
        />
        <path
          d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
        />
        <path
          d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
        />
        <line
          x1="10"
          x2="10"
          y1="11"
          y2="17"
        />
        <line
          x1="14"
          x2="14"
          y1="11"
          y2="17"
        />
      </svg>
    </button>
    <button
      aria-label="Edit"
      class="p-1 mt-2 rounded text-blue-600 hover:text-blue-700 hover:bg-blue-100 transition-colors cursor-pointer"
    >
      <svg
        class="lucide lucide-pen h-3 w-3"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21.174 6.812a1 1 0 0 0-3.986-3.987L3.842 16.174a2 2 0 0 0-.5.83l-1.321 4.352a.5.5 0 0 0 .623.622l4.353-1.32a2 2 0 0 0 .83-.497z"
        />
      </svg>
    </button>
  </div>
  <div
    class="flex-grow py-3 mr-[30px] ml-[30px]"
  >
    <p
      class="text-sm text-gray-900 whitespace-pre-wrap break-words leading-relaxed text-left"
    >
      Test Card Content
    </p>
  </div>
  <div
    class="px-3 pb-2 relative"
  >
    <div
      class="flex items-center justify-between"
    >
      <div
        class="flex items-center"
      >
        <button
          class="p-1 rounded flex items-center transition-colors cursor-pointer text-gray-500 hover:text-green-600 hover:bg-green-50"
        >
          <svg
            aria-label="Upvote"
            class="lucide lucide-thumbs-up h-3 w-3"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 10v12"
            />
            <path
              d="M15 5.88 14 10h5.83a2 2 0 0 1 1.92 2.56l-2.33 8A2 2 0 0 1 17.5 22H4a2 2 0 0 1-2-2v-8a2 2 0 0 1 2-2h2.76a2 2 0 0 0 1.79-1.11L12 2a3.13 3.13 0 0 1 3 3.88Z"
            />
          </svg>
        </button>
        <span
          class="text-xs font-medium mx-1"
        >
          5
        </span>
        <button
          class="p-1 rounded flex items-center transition-colors cursor-pointer text-gray-500 hover:text-red-600 hover:bg-red-50"
        >
          <svg
            aria-label="Downvote"
            class="lucide lucide-thumbs-down h-3 w-3"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 14V2"
            />
            <path
              d="M9 18.12 10 14H4.17a2 2 0 0 1-1.92-2.56l2.33-8A2 2 0 0 1 6.5 2H20a2 2 0 0 1 2 2v8a2 2 0 0 1-2 2h-2.76a2 2 0 0 0-1.79 1.11L12 22a3.13 3.13 0 0 1-3-3.88Z"
            />
          </svg>
        </button>
      </div>
      <div
        class="text-right"
      >
        <div
          class="text-xs text-gray-500 italic"
        >
          Test User
        </div>
      </div>
    </div>
  </div>
</div>
`;
